src/RoslynAnalyzers/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -658,27 +662,29 @@ await checkHelpLinkAsync(helpLinkUri).ConfigureAwait(false))
 
                 return;
 
-                async Task<bool> checkHelpLinkAsync(string helpLink)
+                async Task<(HttpStatusCode, string? responseContent)> checkHelpLinkAsync(string helpLink)
                 {
                     try
                     {
                         if (!Uri.TryCreate(helpLink, UriKind.Absolute, out var uri))
                         {
-                            return false;
+                            // TODO: why would we not fail this if the URI is invalid?
+                            return (HttpStatusCode.OK, null);
                         }
 
                         if (validateOffline)
                         {
-                            return true;
+                            // Just pretend the request worked fine.
+                            return (HttpStatusCode.OK, null);
                         }
 
                         var request = new HttpRequestMessage(HttpMethod.Head, uri);
                         using var response = await httpClient.SendAsync(request).ConfigureAwait(false);
-                        return response?.StatusCode == HttpStatusCode.OK;
+                        return (response.StatusCode, await response.Content.ReadAsStringAsync().ConfigureAwait(false));
                     }
-                    catch (WebException)
```

Comment:
WebException is for WebClient, so this never would have worked.

Reply:
I feel like we definitely must have been getting it, otherwise we wouldnt see it fail like we do

Reply:
So unless you call EnsureSuccessStatusCode it won't throw for many failure types. It's possible it won't throw at all, actually unless the request never could get sent up in the first place.